### PR TITLE
Edge detections for mouse button and key inputs.

### LINKED ISSUE #117: mouse/keyboard edge trigger event
[discord](https://ptb.discord.com/channels/1093814266630512681/1207972838590386187)

reference from [LGF](https://github.com/ntut-xuan/LeistungsstarkesGameFramework/blob/3a433d0262f778ace36fe85694876de3833f6ea6/Source/Game/mygame.h#L76-L92):

```cpp
class CGameStateRun : public CGameState {
public:
        CGameStateRun(CGame *g);
        ~CGameStateRun();
        void OnBeginState();                                                    // 設定每次重玩所需的變數
        void OnInit();                                                                  // 遊戲的初值及圖形設定
        void OnKeyDown(UINT, UINT, UINT);
        void OnKeyUp(UINT, UINT, UINT);
        void OnLButtonDown(UINT nFlags, CPoint point);  // 處理滑鼠的動作
        void OnLButtonUp(UINT nFlags, CPoint point);    // 處理滑鼠的動作
        void OnMouseMove(UINT nFlags, CPoint point);    // 處理滑鼠的動作 
        void OnRButtonDown(UINT nFlags, CPoint point);  // 處理滑鼠的動作
        void OnRButtonUp(UINT nFlags, CPoint point);    // 處理滑鼠的動作
protected:
        void OnMove();                                                                  // 移動遊戲元素
        void OnShow();                                                                  // 顯示這個狀態的遊戲畫面
};
 ```

### PR DESCRIPTION
**Pull request by Dragonfly911117**
Resolves #117

Test with `App::Update()` if you feel like.

### CODE REVIEW

**Review by ntut-rick — COMMENTED** [review]
maybe can merge with keyboard's interface

**Comment by Dragonfly911117** [comment]
> maybe can merge with keyboard's interface

Fixed.

File: include/Util/Keycode.hpp [hunk 1]
```diff
@@ -398,10 +397,19 @@ enum class Keycode {
...
-    PTSD_NUM_SCANCODES = SDL_NUM_SCANCODES /**< not a key, just marks the number of scancodes
+    PTSD_NUM_SCANCODES = SDL_NUM_SCANCODES, /**< not a key, just marks the number of scancodes
                                  for array bounds */
+
+
+    // Below are the keycodes handling mouse buttons.
+
+    PTSD_MOUSE_LB = 513,
+    PTSD_MOUSE_RB = 514,
+    PTSD_MOUSE_MB = 515,
```
Comment (NOOBDY) [inline include/Util/Keycode.hpp:408]: do we need `PTSD_` prefix here since we didn't add it for other keycodes

**Review by NOOBDY — COMMENTED** [review]
IWOMM but small comment

File: include/Util/Keycode.hpp [hunk 2]
```diff
@@ -7,9 +7,8 @@
 
 namespace Util {
 // Prevents from auto format checking fail
-// TODO: Decide whether to reformat this in the future
 // clang-format off
-enum class Keycode {
+enum class  Keycode {
```
[the remark this answers is not in the document]
Comment (Dragonfly911117) [reply include/Util/Keycode.hpp:11]: Forget what was that comment for. That was sth 4 mon. ago.

File: include/Util/Keycode.hpp (hunk 1 above)
Comment (Dragonfly911117) [reply include/Util/Keycode.hpp]: Fixed

**Review by NOOBDY — APPROVED** [review]

**Review by NOOBDY — APPROVED** [review]

**Review by ntut-rick — APPROVED** [review]
lgtm

**Pull request merged**